indico/modules/rb_new/client/js/modules/bookRoom/selectors.js

```diff
@@ -98,13 +100,28 @@ export const isSearchAndPermissionCheckFinished = createSelector(
     (searchFinished, checkingPermissions) => searchFinished && !checkingPermissions
 );
 
+/** Get rooms ids that are unbookable by the user and also take into
+ *  account the maximum advance days property of the room
+ */
+export const getUnbookableRoomIdsWithMaxAdvanceDays = createSelector(
+    userSelectors.getUnbookableRoomIds,
+    roomsSelectors.getAllRooms,
+    getFilters,
+    (unbookableRoomIds, rooms, {dates: {startDate}}) => {
+        const roomIds = Object.entries(rooms).filter(([, room]) => (
```

Comment:
you have to filter out rooms with `!maxAdvanceDays` since that means there's no restriction